### DISCUSSION TITLE
Fanpico board with 4 fan inputs and outputs

### DISCUSSION THREAD

**Discussion by beppo-dd** [tjko/fanpico#10] [Ideas]
@tjko Thanks you for this awesome project 👍 
I want to use it for my modded Fujitsu D538 business PC, which has no fan curve settings. This and all my other PCs have 2–3 fans (incl. power supply fan). So, a 4 fan fanpico would be more than sufficient for me. Therefore, I would like to design a corresponding smaller and cheaper 0404 PCB with 4 fan inputs and 4 fan outputs. The saved 8 pins can then be used for an additional temperature sensor and additional IO interfaces (e.g. for pressure and temperature sensors [BMP280](https://www.lcsc.com/product-detail/Pressure-Sensors_Bosch-Sensortec-BMP280_C83291.html)).
Do you have guidelines for contributions?
Greetings from germany.

**Comment by tjko** [comment]
@beppo-dd,  if you want to create a new board. It would be nice if you'd stick with the existing naming convention so "0404" or "0404D" if you plan on adding display support...

Ideally you should only need to create a new include file for your board in src/boards/ directory (and not need much other changes). But if you want to add new functionality, please try implement it so that it's conditionally only enabled for 'boards' that use it...

0804D.h is probably good starting point, but since you'd only have 4 fans, there likely is no need for multiplexer so could implement fan tachometer reading like in 0804.h (i.e. use "define TACHO_READ_MULTIPLEX 0").

Also, remember that the "MBFAN" PWM read pins need to be assigned to B channel on a 'free' PWM block. And corresponding A channel pins cannot be used for (hardware) PWM, but of course can be used for something else...

Btw, I had ended up creating couple 'custom' KiCad footprints for the floppy power connector and Pico/Pico W compatible footprint (with cutout for the WiFi antenna).  Probably would make sense create some 'library' that could be shared across different board designs?
